apps/angular/interop-rxjs-signal/src/app/list/photos.model.ts

```diff
@@ -0,0 +1,22 @@
+import { Photo } from '../photo.model';
+
+export const STATE_KEYS = ['photos', 'total', 'page', 'term'] as const;
+export type StateKey = (typeof STATE_KEYS)[number];
+
+export type State = {
+  [key in StateKey]: key extends 'photos'
+    ? Photo[]
+    : key extends 'total'
+      ? number
+      : key extends 'page'
+        ? number
+        : key extends 'term'
+          ? string
+          : never;
+};
```

Comment:
Why did you create this State type ? 
Why not creating the interface you really need ?

Reply:
it is my fault. it is automatically generated by vscode. I've fixed it.